akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1322,8 +1322,8 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    *                     on the original, to be emitted in case downstream signals demand.
    * @see [[#expand]]    for a version that can overwrite the original element.
    */
-  def extrapolate[U >: Out](extrapolator: function.Function[U, java.util.Iterator[U]]): javadsl.Flow[In, U, Mat] =
-    new Flow(delegate.extrapolate[U](in ⇒ extrapolator(in).asScala))
+  def extrapolate(extrapolator: function.Function[Out @uncheckedVariance, java.util.Iterator[Out @uncheckedVariance]]): javadsl.Flow[In, Out, Mat] =
```

Comment:
@johanandren : is something like this what you had in mind as a possible solution to the contravariance problem?

Reply:
Yes, that's what I was thinking we need to do.